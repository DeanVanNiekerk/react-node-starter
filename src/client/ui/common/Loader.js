import React from 'react'

const Loader = ({ text = "loading..." }) => {

    return <div className="mt-5 mb-5">
        <div className="row justify-content-center">
            {text}
        </div>
    </div>

}

export default Loader