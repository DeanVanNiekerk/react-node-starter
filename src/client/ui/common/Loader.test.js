import React from 'react';
import Loader from './Loader';
import renderer from 'react-test-renderer';

describe('Loader', () => {

    it('renders correctly - with no text', () => {

        const tree = renderer
            .create(<Loader />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly - with text', () => {

        const tree = renderer
            .create(<Loader text="Custom loader message" />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

});