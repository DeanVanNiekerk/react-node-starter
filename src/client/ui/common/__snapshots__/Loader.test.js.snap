// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly - with no text 1`] = `
<div
  className="mt-5 mb-5"
>
  <div
    className="row justify-content-center"
  >
    loading...
  </div>
</div>
`;

exports[`Loader renders correctly - with text 1`] = `
<div
  className="mt-5 mb-5"
>
  <div
    className="row justify-content-center"
  >
    Custom loader message
  </div>
</div>
`;
